src/sltyledComponents/card.js

```diff
@@ -0,0 +1,164 @@
+        transition: all .3s linear;
+    }
+
+    div {
+        flex: 1;
+        text-align: center;
+        text-transform: uppercase;
+        display: flex;
+        align-items: center;
+        justify-content: center;
+    }
+
+    display: flex;
+    font-weight: bold;
+    text-align: center;
+    position: relative;
+    font-size: 2rem;
+    flex: 1;
+    border-radius: 0 0 0 18px;
+`;
+
+const CardImage = styled.img`
+    width: 100%;
+    height: 242px;
+`;
+
+const ImageContainer = styled.div`
+    width: 100%;
+    height: 100%;
+    position: relative;
+    overflow: hidden;
+
+    &::before {
+        content: " ";
+        position: absolute;
+        width: 100%;
+        height: 100%;
+        z-index: 1;
+        transform: scale(0);
+        background: rgba(5, 66, 97, 0.65);
+        transform-origin: center;
+    }
+
+    &::after {
+        content: " ";
+        position: absolute;
+        left: 0;
+        z-index: 1;
+        border: 2px white solid;
+        border-radius: 5px;
+        transform-origin: center;
+    }
+`;
+
+const CardContainer = styled.div`
+    display:flex;
+    flex-direction:column;
+    min-width: 230px;
+    max-width: 300px;
+    height: 300px;
+    margin: 30px;
+    flex: 1;
+    z-index: 5;
+    border-radius: 0 0 0 25px;
+    overflow: hidden;
+    border: 4px solid ${({color}) => color ? color : "red"};
+    background-color: white;
+    opacity: 1;
+    //box-shadow: 0 2px 5px 0 rgba(0,0,0,0.16), 0 2px 10px 0 rgba(0,0,0,0.12);
+    box-shadow: 0px 6px 6px 4px rgba(0, 0, 0, 0.70);
+    cursor: pointer;
+    transition: transform .5s ease-out;
+
+    &:hover {
+        transform: scale(1.2);
+    }
+
+    ${CardName} {
+        background-color: ${({color}) => color ? color : "red"};
+        border: 4px ${({color}) => color ? color : "red"} solid;
+    }
+
+    &:hover ${CardName} {
+        color: ${({color}) => color ? color : "red"};
+        background-color: white;
+        border-top: 4px ${({color}) => color ? color : "red"} solid;
+    }
+
+    &:hover ${CardImage} {
+        transition: transform ease-in .6s;
+        transform:scale(1.2);
+    }
+
+    &:hover ${ImageContainer} {
+        &::before {
+            transform: scale(1);
+            transition: all 0.2s ease-in;
+        }
+
+        &::after {
+            left: 10px;
+            top: 10px;
+            right: 10px;
+            bottom: 10px;
+            animation: ${rotate} .6s linear 1;
+        }
+    }
+`;
+
+const getPokemonIdByUrl = (url) => {
+    let urlArray = url.split('/');
+    return urlArray[urlArray.length - 2];
+}
+
+function Card(props) {
+    let history = useHistory();
+    const cardClick = (id) => {
+        history.push(`/details/${id}`);
+    }
+    const { color, pokemon } = props;
+    const pokemonId = getPokemonIdByUrl(pokemon.url);
+    const imageUrl = `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/${pokemonId}.png`;
```

Comment:
This is a logic you have figure it out, maybe documented somewhere, else for the information about sprites you ned to fetch details about a specific pokemon: https://pokeapi.co/docs/v2#pokemon

Your solution in this case is not very safe but has the advantage of removing cascade of API calls on list page.

Reply:
Yes, it's just like for the list I just needed the image, and making a api call for each of the pokemons in the list felt like a bad idea, you're right, I should've documented it.